pkg/cmd/kafka/acl/create/create.go

```diff
@@ -53,12 +55,26 @@ func NewCreateCommand(f *factory.Factory) *cobra.Command {
 				return flag.RequiredWhenNonInteractiveError("yes")
 			}
 
-			if err := aclutil.ValidateAndSetResources(opts, flagutil.ResourceTypeFlagEntries); err != nil {
-				return err
+			var errorMsg string = opts.Localizer.MustLocalize("common.error.invalidFlagCombination")
+
+			if opts.Permission == "" {
+				errorMsg = aclutil.BuildBulletPoints(errorMsg, opts.Localizer.MustLocalize("kafka.acl.common.flag.permission.required"))
+			}
+
+			if opts.Operation == "" {
+				errorMsg = aclutil.BuildBulletPoints(errorMsg, opts.Localizer.MustLocalize("kafka.acl.common.flag.operation.required"))
+			}
+
+			if resourceErrors := aclutil.ValidateAndSetResources(opts, flagutil.ResourceTypeFlagEntries); resourceErrors != "" {
+				errorMsg = aclutil.BuildBulletPoints(errorMsg, resourceErrors)
+			}
+
+			if principalErrors := validateAndSetOpts(opts); principalErrors != "" {
+				errorMsg = aclutil.BuildBulletPoints(errorMsg, principalErrors)
 			}
 
-			if err := validateAndSetOpts(opts); err != nil {
-				return err
+			if errorMsg != opts.Localizer.MustLocalize("common.error.invalidFlagCombination") {
+				return errors.New(errorMsg)
 			}
```

Comment:
We should not be comparing the string content of an error message, refer to other comments about using `error` type where possible.